File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    answer = ''
    
    dictionary = dict()
    for c in participant :
        if c in dictionary :
            dictionary[c] += 1
        else :
            dictionary[c] = 1
            
    #print(dictionary)
    for c in completion :
        dictionary[c] -= 1
    #print(dictionary)
    for key, value in dictionary.items() :
        if value >= 1 :
            return key
        
    
    return answer

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;

class Solution {
    public String solution(String[] participant, String[] completion) {
        String answer = "";
        HashMap<String, Integer> hashmap = new HashMap<>();
        
        for (String c : completion) {
            if (hashmap.containsKey(c)) {
                int cnt = hashmap.get(c);
                hashmap.put(c, cnt+1);
            } else {
                hashmap.put(c, 1);
            }
        }
        
        // System.out.println(hashmap);
        
        for (String p : participant) {
            if (!hashmap.containsKey(p)) {
                answer = p;
                break;
            }
            int cnt = hashmap.get(p);
            if (cnt == 0) {
                answer = p;
                break;
            }
            hashmap.put(p, cnt-1);
        }
        
        return answer;
    }
}